opengever/maintenance/scripts/set_iworks_mimetypes.py

```diff
@@ -0,0 +1,50 @@
+import mimetypes
+from opengever.maintenance.debughelpers import setup_app
+from opengever.maintenance.debughelpers import setup_option_parser
+from opengever.maintenance.debughelpers import setup_plone
+from os.path import splitext
+from plone import api
+import transaction
+
+
+def set_iworks_mimetype():
+    """Search all MS OneNote documents and fixes the contentType.
+    """
+
+    catalog = api.portal.get_tool('portal_catalog')
+    brains = catalog.unrestrictedSearchResults(
+        portal_type='opengever.document.document')
+    # A getIcon value of 'application.png' indicates that the mimetype
+    # coulnd't be resolved for this document - only consider these
+    brains = filter(lambda b: b.getIcon == 'application.png', brains)
+    for brain in brains:
+        obj = brain.getObject()
+        fname = obj.file.filename
+        ext = splitext(fname)[1]
+        if ext in [".numbers", ".pages", ".key"]:
+            obj.file.contentType = mimetypes.guess_type(fname)[0]
+            obj.reindexObject()
+            print u'File {} with {} fixed.'.format(obj, obj.file.filename)
```

Comment:
i'd probably omit printing each file separately here and just be happy with what `ProgressLogger` tells us. (Also it would be nicer to use `logging` in such cases, but here i'd just drop it 😉 )